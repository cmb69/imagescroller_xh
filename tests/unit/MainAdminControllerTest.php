<?php

/*
 * Copyright 2023 Christoph M. Becker
 *
 * This file is part of Imagescroller_XH.
 *
 * Imagescroller_XH is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * Imagescroller_XH is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with Imagescroller_XH.  If not, see <http://www.gnu.org/licenses/>.
 */

namespace Imagescroller;

use ApprovalTests\Approvals;
use Imagescroller\Infra\Repository;
use Imagescroller\Infra\View;
use Imagescroller\Value\Image;
use PHPUnit\Framework\TestCase;

class MainAdminControllerTest extends TestCase
{
    public function testRendersSelectboxAndGallery(): void
    {
        $_GET = ["imagescroller_gallery" => "gallery2"];
        $repository = $this->createMock(Repository::class);
        $repository->method("find")->willReturn($this->images());
        $repository->method("findAll")->willReturn(["gallery1", "gallery2", "gallery3"]);
        $view = new View("./views/", XH_includeVar("./languages/en.php", "plugin_tx")["imagescroller"]);
        $sut = new MainAdminController($repository, $view);
        ob_start();
        $sut->defaultAction();
        $response = ob_get_clean();
        Approvals::verifyHtml($response);
    }

    private function images(): array
    {
        return [
            new Image("image1"),
            new Image("image2"),
            new Image("image3"),
        ];
    }
}
