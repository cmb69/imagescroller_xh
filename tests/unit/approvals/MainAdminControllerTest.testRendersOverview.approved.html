<!-- imagescroller overview -->
<section id="imagescroller_admin">
  <h1>Imagescroller â€“ Galleries</h1>
  <form method="get">
    <input type="hidden" name="selected" value="imagescroller">
    <input type="hidden" name="admin" value="plugin_main">
    <table>
      <tr>
        <td><input type="radio" name="imagescroller_gallery" value="gallery1"></td>
        <td>gallery1</td>
      </tr>
      <tr>
        <td><input type="radio" name="imagescroller_gallery" value="gallery2"></td>
        <td>gallery2</td>
      </tr>
      <tr>
        <td><input type="radio" name="imagescroller_gallery" value="gallery3"></td>
        <td>gallery3</td>
      </tr>
    </table>
    <p>
      <button name="action" value="create">Create gallery from folder</button>
    </p>
  </form>
</section>
